shopping-webclient/src/components/checkout/stripe/Stripe.vue

```diff
@@ -52,15 +62,94 @@ export default {
   },
 
   methods: {
-    async pay() {
+    async pay(event) {
       // createToken returns a Promise which resolves in a result object with
       // either a token or an error key.
       // See https://stripe.com/docs/api#tokens for the token object.
       // See https://stripe.com/docs/api#errors for the error object.
       // More general https://stripe.com/docs/stripe.js#stripe-create-token.
       try {
-        const data = await Stripe.createToken();
-        this.$emit('pay', data.token);
+        // const data = await Stripe.createToken();
+        // this.$emit('pay', data.token);
+        event.target.disabled = true;
+        let checkoutId = this.checkoutId;
+        const { data } = await Vue.prototype.$axios({
+          url: ProxyUrl.stripeInstantPay,
+          method: 'post',
+          data: {
+            checkoutId,
+          },
+        });
+        if (data && data.httpStatus === 200) {
+          console.log(data.responseData);
+          if (data.responseData) {
+            const paymentMethodReq = await Stripe.createPaymentMethod(
+              'card',
+              {}
+            );
+            console.log(paymentMethodReq);
+
+            if (paymentMethodReq.paymentMethod.id) {
+              const confirmPayment = await Stripe.confirmCardPayment(
+                data.responseData.client_secret,
+                {
+                  payment_method: paymentMethodReq.paymentMethod.id,
+                }
+              );
+
+              console.log(confirmPayment);
+
+              if (confirmPayment) {
+                console.log('inside confirmPayment', confirmPayment);
+                if (confirmPayment.paymentIntent.status == 'succeeded') {
+                  console.log('inside confirmPayment status', confirmPayment);
+                  if (confirmPayment.hasOwnProperty('error')) {
+                    this.$notify({
```

Comment:
If we go into this error condition, shouldn't we return from the function at the end? Or displaying the error and then making `ProxyUrl.stripeInstantPayment` call the desired behavior?

Reply:
You need not return a function. I've tested various Stripe errors with this implementation and I've had no issues with the current error handling.